### PR TITLE
Add cmds/exp/vmboot as experimental bootloader

### PR DESCRIPTION
**Pull request by ChriMarMe**
vmboot utilizes gokvm to instantiate a linux kvm and executes EDK2/CloudHV from a drive attached to the system.

only viable on 
- x86-systems with AMD-V or Intel-VT support
- Linux kernel build with kvm support

### CODE REVIEW

**Comment by rminnich** [comment]
It seems to me this would be a tidy. compact, system if you avoided the vendoring. I am wondering why that is needed, given that we might use Go modules. Am I missing something? This is quite neat.

**Comment by ChriMarMe** [comment]
> It seems to me this would be a tidy. compact, system if you avoided the vendoring. I am wondering why that is needed, given that we might use Go modules. Am I missing something? This is quite neat.

I wondered as well. Adding gokvm via `go get github.com/bobuhiro11/gokvm@main` 
then ` go build .` had errors with multiple dependencies marked as required by go.mod, but not in vendor.
`go mod tidy` then `go mod vendor` fixed the issue and I could build.

I cant figure out the hoops I have to jump through to avoid vendoring.....annoying

**Comment by rminnich** [comment]
I will look at this I guess, I'd really rather not vendor all that stuff in.

**Comment by rminnich** [comment]
go build -mod=mod

works fine. So what I'd suggest for now is that we bring this in, with a warning that it can't quite build into u-root yet? 
Chris Koch, do you think we can ever remove vendor/ ?

**Comment by rminnich** [comment]
but yeah, guess this means go test ./... fails. Hmm.

**Comment by rminnich** [comment]
ah, never mind, let's go with this and go with the vendoring. If you can fix the conflicts it works for me. Thanks. 

I really wish we could drop vendor/ but it seems not quite yet.

**Comment by ChriMarMe** [comment]
I have no clue, why build_and_test fails. 

The Inst-function is in `debug_amd64.go`, but the build process does not find it. Why?

**Comment by ChriMarMe** [comment]
Is there a way to exclude ARM build tests for this in particular? ARM simply is not supported by gokvm, hence the required function Inst is never found in the code for any arm target.

Maybe you have other ideas to make the CI happy

**Comment by ChriMarMe** [comment]
> Is there a way to exclude ARM build tests for this in particular? ARM simply is not supported by gokvm, hence the required function `Inst` is never found in the code for any arm target.
> 
> Maybe you have other ideas to make the CI happy

Fixed it I guess. :dagger:

**Comment by ChriMarMe** [comment]
Configuration of the virtual machine is pretty static at the moment, but it runs EDK2/CloudHv until EFI-Shell.

This shall serve as attraction point to get peoples interest. Improvements in gokvm are required for it to really be useful in any way. I hope poeple like the idea of running a vm from u-root. :)

**Comment by rminnich** [comment]
do you want to convert this from draft? Why don't we just get it in?

**Comment by ChriMarMe** [comment]
Done

**Comment by MDr164** [comment]
Maybe change "WIP" to "experimental", otherwise looks pretty good!

File: cmds/exp/vmboot/vmboot_amd64.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		if err != nil {
+			return err
+		}
+		mp.Add(m)
+	}
+
+	return nil
+}
+
+func filterNonPartiotions(devs block.BlockDevices) block.BlockDevices {
```
Comment (MDr164) [inline cmds/exp/vmboot/vmboot_amd64.go:105]: Typo: filterNonPartitions

Comment (ChriMarMe) [reply cmds/exp/vmboot/vmboot_amd64.go]: fixed

**Review by MDr164 — APPROVED** [review]

**Pull request merged**